src/docfx/lib/moniker/MonikerRangeParser.cs

```diff
@@ -2,20 +2,39 @@
 // Licensed under the MIT license. See LICENSE file in the project root for full license information.
 
 using System.Collections.Generic;
+using System.IO;
 
 namespace Microsoft.Docs.Build
 {
     internal class MonikerRangeParser
     {
         private readonly EvaluatorWithMonikersVisitor _monikersEvaluator;
 
-        public MonikerRangeParser(IEnumerable<Moniker> monikers)
+        public MonikerRangeParser(MonikerDefinitionModel monikerDefinition)
         {
-            _monikersEvaluator = new EvaluatorWithMonikersVisitor(monikers);
+            if (monikerDefinition == null)
+            {
+                _monikersEvaluator = null;
+            }
+            _monikersEvaluator = new EvaluatorWithMonikersVisitor(monikerDefinition);
+        }
+
+        public static (List<Error>, MonikerRangeParser) Create(Context context, string monikerDefinitionFile)
+        {
+            if (string.IsNullOrEmpty(monikerDefinitionFile))
+            {
+                return (new List<Error>(), new MonikerRangeParser(null));
+            }
+            var (errors, monikerDefinition) = JsonUtility.Deserialize<MonikerDefinitionModel>(File.ReadAllText(monikerDefinitionFile));
```

Comment:
Are we going to have additional properties other than `List<MonikerSpec> Monikers`?
If not, we can we just use
```csharp
JsonUtility.Deserialize<List<MonikerSpec>>
```

Reply:
I think we should keep the Scalability just like `XrefMapModel`

---
In reply to: [230628316](https://github.com/dotnet/docfx/pull/3575#discussion_r230628316) [](ancestors = 230628316)